CadbaseLibrary/CadbaseLibrary.FCMacro

```diff
@@ -0,0 +1,837 @@
+
+        if reply.error() == QtNetwork.QNetworkReply.NoError:
+
+            responseData = reply.readAll()
+
+            with open(filepath, 'wb') as f:
+                f.write(responseData)
+
+            return (filepath, time.time() - t0)
+        else:
+            FreeCAD.Console.PrintWarning('Error')
+    except Exception as e:
+        FreeCAD.Console.PrintWarning('Exception in download file: ')
+        FreeCAD.Console.PrintWarning(e)
+
+class CDBSApi:
+
+    def __init__(self, query, LIBRARYPATH):
+
+        FreeCAD.Console.PrintWarning('Getting data...\n')
+        self.doRequest(query, LIBRARYPATH)
+
+
+    def doRequest(self, query, LIBRARYPATH):
+
+        api_url = param.GetString('api-url', '')
+        req = QtNetwork.QNetworkRequest(QtCore.QUrl(api_url))
+
+        auth_header = 'Bearer ' + param.GetString('auth-token', '')
+        header = {'Authorization': auth_header}
+
+        req.setRawHeader(b"Content-Type", content_type);
+        req.setRawHeader(b"Authorization", auth_header.encode());
+
+        body = json.dumps(query).encode('utf-8')
+
+        self.nam = QtNetwork.QNetworkAccessManager()
+        reply = self.nam.post(req, body)
+        loop = QtCore.QEventLoop()
+        reply.finished.connect(loop.quit)
+        loop.exec_()
+        self.handleResponse(reply, LIBRARYPATH)
+
+    def handleResponse(self, reply, LIBRARYPATH):
+
+        new_info_file = LIBRARYPATH + "/" + RESPONSEFILE
+
+        if os.path.exists(new_info_file):
+            if os.path.isdir(new_info_file):
+                os.rmdir(new_info_file)
+            else:
+                os.remove(new_info_file)
+
+        er = reply.error()
+
+        if er == QtNetwork.QNetworkReply.NoError:
+            if reply.attribute(QtNetwork.QNetworkRequest.HttpStatusCodeAttribute) == 200:
+
+                bytes_string = reply.readAll()
+                # FreeCAD.Console.PrintWarning(bytes_string)
+                with open(new_info_file, 'wb') as file:
+                    file.write(bytes_string)
+
+                FreeCAD.Console.PrintWarning('Success\n')
+
+            else:
+                FreeCAD.Console.PrintWarning('Failed, status code: ')
+                FreeCAD.Console.PrintWarning(reply.attribute(QtNetwork.QNetworkRequest.HttpStatusCodeAttribute))
+
+        else:
+            FreeCAD.Console.PrintWarning("Error occured: ")
+            FreeCAD.Console.PrintWarning(er)
+            FreeCAD.Console.PrintWarning('\n')
+            FreeCAD.Console.PrintWarning(reply.errorString())
+            FreeCAD.Console.PrintWarning('\n')
+
+def parsingGpl(LIBRARYPATH):
+    FreeCAD.Console.PrintWarning('Data processing, please wait.\n')
+    response_path = LIBRARYPATH + "/" + RESPONSEFILE
+    # FreeCAD.Console.PrintWarning(response_path + '\n')
+
+    if os.path.exists(response_path):
+
+        f = open(response_path, "rb", buffering=0)
+
+        x = json.loads(f.readall(), object_hook=lambda d: SimpleNamespace(**d))
+
+        if x.data == None:
```

Comment:
Use `is` to compare with None or `if x.data`, if possible.

Reply:
Corrected.